src/modules/27-platform/secrets/components/SecretReference/__test__/SecretReference.test.tsx

```diff
@@ -6,70 +6,291 @@
+    const accountTab = screen.getByText('account')
+    expect(projectTab).toBeInTheDocument()
+    expect(orgTab).toBeInTheDocument()
+    expect(accountTab).toBeInTheDocument()
+    // Check if secret from response is appearing on screen
+    const secretName = screen.getByText('text1')
+    expect(secretName).toBeInTheDocument()
   })
+
   test('render with spinner', async () => {
+    const listSecretsV2PromiseMock = jest.fn(() =>
+      Promise.resolve({
+        loading: true,
+        error: null,
+        data: null,
+        refetch: jest.fn().mockReturnValue(null)
+      })
+    )
+    ;(listSecretsV2Promise as jest.Mock).mockImplementation(listSecretsV2PromiseMock)
+
     const { container } = render(
       <TestWrapper>
-        <SecretReference accountIdentifier="dummy" mock={mockData as any} onSelect={noop} />
+        <SecretReference accountIdentifier="dummy" onSelect={noop} />
       </TestWrapper>
     )
     expect(getByText(container, 'account')).toBeTruthy()
-    expect(queryByAttribute('data-icon', container, /spinner/)).toBeTruthy()
-    expect(container).toMatchSnapshot()
+    expect(queryByAttribute('data-icon', container, /spinner/)).toBeInTheDocument()
+    const secretName = screen.queryByText('text1')
+    expect(secretName).not.toBeInTheDocument()
   })
+
   test('render with no data', async () => {
-    mockData.data.empty = true
-    mockData.data.content = []
+    const listSecretsV2PromiseMock = jest.fn(() =>
+      Promise.resolve({
+        loading: false,
+        error: null,
+        data: { data: { content: [] } },
+        refetch: jest.fn().mockReturnValue(null)
+      })
+    )
+    ;(listSecretsV2Promise as jest.Mock).mockImplementation(listSecretsV2PromiseMock)
+
     const { container } = render(
       <TestWrapper>
-        <SecretReference accountIdentifier="dummy" mock={mockData as any} onSelect={noop} />
+        <SecretReference accountIdentifier="dummy" onSelect={noop} />
       </TestWrapper>
     )
     await waitFor(() => getByText(container, 'entityReference.apply'))
-    expect(getByText(container, 'platform.secrets.secret.noSecretsFound')).toBeTruthy()
+    expect(getByText(container, 'platform.secrets.secret.noSecretsFound')).toBeInTheDocument()
     await waitFor(() => {
       // for SelectTypeDropdown (text/file)
       expect(getByText(container, 'platform.secrets.secret.labelSecretType')).toBeInTheDocument()
     })
-    expect(container).toMatchSnapshot()
+
+    const secretName = screen.queryByText('text1')
+    expect(secretName).not.toBeInTheDocument()
+  })
+
+  test('it should call secrets v2 api call with secretManagerIdentifiers query param when connectorTypeContext is of secret manager type', async () => {
+    const listSecretsV2PromiseMock = jest.fn(() =>
+      Promise.resolve({
+        loading: false,
+        error: null,
+        data: mockData.data,
+        refetch: jest.fn().mockReturnValue(mockData.data)
+      })
+    )
```

Comment:
You can simplify with `mockResolvedValue`
```suggestion
    const listSecretsV2PromiseMock = jest.fn().mockResolvedValue({
      loading: false,
      error: null,
      data: mockData.data,
      refetch: jest.fn().mockReturnValue(mockData.data)
    })
```

Reply:
Sure, Kevin. Noted.